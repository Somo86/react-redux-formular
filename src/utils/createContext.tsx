import * as React from 'react';
import {Context} from 'react';

//** It Produces similar behaviour as React Context
//** additionally it allows to be used in older React versions

export default function createContext<T>(defaultValue: Object = {}): Context<T> {

  let state = defaultValue ;

  function ConsumerComponent(props): JSX.Element {
    const newProps = {...props, ...state};
    return (
      <div>
        { props.children(newProps) }
      </div>
    )
  }

  function ProviderComponent(props): JSX.Element {
    state = props.value;
    return (
      <div>
        { props.children }
      </div>
    );
  }

  return {
    Provider: ProviderComponent,
    Consumer: ConsumerComponent,
  }
}
