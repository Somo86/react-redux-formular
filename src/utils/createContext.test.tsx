import * as React from 'react';
import * as renderer from 'react-test-renderer';
import createContext from './createContext';

describe('create context component', () => {
  const ContextComponent = createContext({});
  const providerComponent = renderer.create(
    <ContextComponent.Provider value={{state: 'test'}}>
      <div>
        <input />
      </div>
    </ContextComponent.Provider>
  );

  /*const consumerComponent = renderer.create(
    <ContextComponent.Consumer>
      {(context) => (
        <p>test</p>
      )}
    </ContextComponent.Consumer>
  );*/

  test('should return a Provider nd a Consumer', () => {
    expect(Object.keys(ContextComponent)).toContain('Provider');
    expect(Object.keys(ContextComponent)).toContain('Consumer');
  });

  test('',() => {
    expect(providerComponent.toJSON()).toMatchSnapshot();
  })
});