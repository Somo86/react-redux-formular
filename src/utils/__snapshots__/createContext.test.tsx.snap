// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create context component  1`] = `
<div>
  <div>
    <input />
  </div>
</div>
`;
